### PR TITLE
Iss252

### PR DESCRIPTION
**Pull request by burkmarr**
Change to dynamic_location form to remove map editing controls and submission buttons if the user does not own the location (unless admin). For https://github.com/BiologicalRecordsCentre/iRecord/issues/252.

### CODE REVIEW

File: prebuilt_forms/dynamic_location.php [hunk 1]
```diff
@@ -303,9 +303,15 @@ protected static function get_control_map($auth, $args, $tabalias, $options) {
     if ($tabalias)
       $options['tabDiv'] = $tabalias;
     $olOptions = iform_map_get_ol_options($args);
+    if (!self::userOwnsLocation()) {
+      $options['standardControls']=null;
```
Comment (johnvanbreda) [inline prebuilt_forms/dynamic_location.php]: Drupal coding standards - space around "=" and capitalise NULL

File: prebuilt_forms/dynamic_location.php [hunk 2]
```diff
@@ -303,9 +303,15 @@ protected static function get_control_map($auth, $args, $tabalias, $options) {
...
+      $r = '<p><strong>You cannot edit this loation because you do not own it.</strong></p>';
```
Comment (johnvanbreda) [inline prebuilt_forms/dynamic_location.php]: Typo loation

File: prebuilt_forms/dynamic_location.php [hunk 3]
```diff
@@ -556,5 +565,22 @@ protected static function getSubmitButtons($args) {
     return $r;
   }
 
+  private function userOwnsLocation() {
```
Comment (johnvanbreda) [inline prebuilt_forms/dynamic_location.php]: Coding standards - function comment header missing.

File: prebuilt_forms/dynamic_location.php [hunk 4]
```diff
@@ -556,5 +565,22 @@ protected static function getSubmitButtons($args) {
...
+    if (empty(data_entry_helper::$entity_to_load['location:id'])) {
+      return true;
```
Comment (johnvanbreda) [inline prebuilt_forms/dynamic_location.php]: Coding standards - capitalise TRUE and FALSE throughout this function.

**Review by johnvanbreda — COMMENTED** [review]
There may be more logic to add to this in future - for example if a site is owned by a single website and the user is a site admin (in websites_users table on warehouse) then they should probably be allowed to edit.

File: prebuilt_forms/dynamic_location.php (hunk 3 above)
Comment (burkmarr) [reply prebuilt_forms/dynamic_location.php]: Also renamed function to better reflect what it does.

**Comment by burkmarr** [comment]
Thanks @johnvanbreda - I've addressed those issues.

**Pull request merged**